### DISCUSSION TITLE
reading in .shp files

### DISCUSSION THREAD

**Discussion by robinsandfort** [weecology/DeepForest#781] [Q&A] [answered]
Hi everyone,
I struggle reading in shape files:
with
shp = utilities.read_file(input="{}/test_read_file_boxes_projected.shp".format(tmpdir))

I used:
shape = get_data("shape//DJI_20240911092253_0119_Z.shp")
shp = utilities.read_file(input=shape)
and
shp = utilities.read_file(shape)

I do have an image_path column and I tried different ways to point it to the image position.
It is in the same folder as the shapefile

The output is:
Found image_path column in shapefile, using shape/DJI_20240911092253_0119_Z.JPG
Geometry type of shapefile is Polygon

but also:

CPLE_OpenFailedError: shape/DJI_20240911092253_0119_Z.JPG: No such file or directory

Any ideas?
Greetings,
Robin

**Comment by ethanwhite** [comment, marked as answer]
Hi @robinsandfort - `read_data()` is taking a file path as input. So you want something like

```python
# Full path version
 shp = utilities.read_file(input="/full/path/to/test_read_file_boxes_projected.shp")

# Relative path version
 shp = utilities.read_file(input="path/to/test_read_file_boxes_projected.shp")
```

`get_data()` is only for access data we distribute with the package for testing/examples.

The stuff with `{}` and `.format(tmpdir)` is because this docs example was copied from our tests. We'll put it on the list to clean that up since it's definitely confusing.

**Reply by ethanwhite** [reply]
I've opened a PR to make these docs easier to understand https://github.com/weecology/DeepForest/pull/783
